materialize-mysql/driver.go

```diff
@@ -637,12 +665,14 @@ func (d *transactor) Load(it *pm.LoadIterator, loaded func(int, json.RawMessage)
 			}
 
 			batches[it.Binding] = append(batches[it.Binding], converted...)
+			batchSize += size.Of(converted)
```

Comment:
I'm not terribly excited about the new dependency here and the runtime inspection of a variable's size. Since we are building these converted values into strings to write to a CSV file in the `drainBatch` functions, I think we could re-arrange things so a batch is a `[]string`, and the sizing could be based on the byte length of those strings.

More broadly, have you considered streaming directly to the reader? Squinting at the code here a little bit I think you could set up an `io.Pipe()` that would send rows directly through to the reader in `mysql.RegisterReaderHandler`. The CSV writer would write to the write end of the pipe, and the "`ReaderHandler`" would then read from the read end.

Reply:
That's actually interesting, I think now that we don't need the arguments separately anymore, we could do something like that, to use a `bytes.Buffer` to hold the CSV encoding of each batch and then measuring the size of that is also much easier 🤔

I'll look into this, thanks Will